app/javascript/components/DownloadButton.js

```diff
@@ -29,47 +25,61 @@ function fetchDownloadConfig() {
     'curl -K cfg.txt; rm cfg.txt'
   );
 
-  return [authCode, timeInterval, downloadCommand];
+  return {
+    authCode: authCode,
+    timeInterval: timeInterval,
+    downloadCommand: downloadCommand
+  };
 }
 
 function DownloadCommandContainer() {
-  
-  const [authCode, timeInterval, downloadCommand] = fetchDownloadConfig();
 
-  const expiresMinutes = Math.floor(timeInterval / 60); // seconds -> minutes
+  const [downloadConfig, setDownloadConfig] = useState({});
 
-  const commandID = 'command-' + authCode;
+  useEffect(() => {
+    const fetchData = async () => {
+      const dlConfig = await fetchDownloadConfig();
+      setDownloadConfig(dlConfig);
+    };
+    fetchData();
+  }, []);
+
+  function onSuccess() {
```

Comment:
Is there a ticket for this TODO?

Reply:
Addressed in 034ff9d0.  SCP-2144 tracks linting for such issues.